azure/scope/cluster.go

```diff
@@ -266,20 +267,18 @@ func (s *ClusterScope) NSGSpecs() []azure.NSGSpec {
 }
 
 // SubnetSpecs returns the subnets specs.
-func (s *ClusterScope) SubnetSpecs() []azure.SubnetSpec {
-	numberOfSubnets := len(s.AzureCluster.Spec.NetworkSpec.Subnets)
-	if s.AzureCluster.Spec.BastionSpec.AzureBastion != nil {
-		numberOfSubnets++
-	}
-
-	subnetSpecs := make([]azure.SubnetSpec, 0, numberOfSubnets)
+func (s *ClusterScope) SubnetSpecs() []azure.ResourceSpecGetter {
+	subnetSpecs := []azure.ResourceSpecGetter{}
```

Comment:
It looks like you know the number of specs you are going to create. Perhaps, preallocate the slice using `make([]azure.ResourceSpecGetter, len(s.AzureCluster.Spec.NetworkSpec.Subnets))` or `make([]azure.ResourceSpecGetter, 0, len(s.AzureCluster.Spec.NetworkSpec.Subnets) + 1)`.